### DISCUSSION TITLE
Poor Wifi and Ethernet connectivity

### DISCUSSION THREAD

**Discussion by AlienKevin** [macOS-Dell-Latitude-E7470/macOS-Dell-Latitude-E7470#35] [Q&A] [answered]
My Wifi connection drops quite often and has low speed even when connected. To ensure that my home internet is good, I tested it with my phone at around the same time and the connection is about 18mb/s. I then tried to use the Ethernet port instead. Now my internet connection does not drop, but it is still very slow (~300kb/s). Here's my ping statistics with Ethernet:
```
--- www.bing.com ping statistics ---
13 packets transmitted, 9 packets received, 30.8% packet loss
round-trip min/avg/max/stddev = 1084.995/1755.651/2262.247/410.283 ms
```
To upgrade OpenCore, I tried copying the latest EFI folder in this repo to the EFI folder on my hard drive, but it doesn't seem to change the version number obtained by `nvram 4D1FDA02-38C7-4A6A-9CC6-4BCCA8B30102:opencore-version`.

My machine:
```
OC: REL-064-2020-12-07
MacOS: 11.3.1
```

**Comment by adityabakare** [comment]
> My Wifi connection drops quite often and has low speed even when connected. To ensure that my home internet is good, I tested it with my phone at around the same time and the connection is about 18mb/s. I then tried to use the Ethernet port instead. Now my internet connection does not drop, but it is still very slow (~300kb/s). Here's my ping statistics with Ethernet:
> 
> ```
> --- www.bing.com ping statistics ---
> 13 packets transmitted, 9 packets received, 30.8% packet loss
> round-trip min/avg/max/stddev = 1084.995/1755.651/2262.247/410.283 ms
> ```

itlwm is the kext to blame for poor Wi-Fi connectivity. Upgrade to latest stable kext from OpenIntelWireless and you'll be fine(The latest commit already has it). Not sure about Ethernet though. It's quite stable for me.

> 
> To upgrade OpenCore, I tried copying the latest EFI folder in this repo to the EFI folder on my hard drive, but it doesn't seem to change the version number obtained by `nvram 4D1FDA02-38C7-4A6A-9CC6-4BCCA8B30102:opencore-version`.
> 
> My machine:
> 
> ```
> OC: REL-064-2020-12-07
> MacOS: 11.3.1
> ```

I've already added the upgrade instructions to [README.md](https://github.com/adityabakare/macOS-Dell-Latitude-E7470#important-opencore-066-update-and-newer). You need to go through them if you are on OC 0.6.5 or older.

**Comment by AlienKevin** [comment]
I followed the steps on the [dortana website](https://dortania.github.io/OpenCore-Post-Install/multiboot/bootstrap.html#updating-bootstrap-in-0-6-6) to upgrade from OC 0.6.5 to the latest version. However, the version number I got is still the same. Wonder if I should not be running `nvram 4D1FDA02-38C7-4A6A-9CC6-4BCCA8B30102:opencore-version`?

**Comment by adityabakare** [comment, marked as answer]
If I recall correctly, you need to Reset your NVRAM, then point the OpenCore.efi correctly in the bootmenu after that.

Moving to Discussions

**Reply by AlienKevin** [reply]
I reset the NVRAM by on the start up menu. But how can I point the OpenCore.efi in the bootmenu after that?

**Reply by adityabakare** [reply]
Go to Bios Setup > General Settings > Boot Sequence and add a boot option from there

**Reply by AlienKevin** [reply]
Thanks for your help. During bootup, the OpenCore version is 2021-04-05 so I think I managed to upgrade! I think the problem might be the way I test OC version in ther Terminal. It's still says the old version (2020-12-07) for some reason.
